lib/graphql_schema.rb

```diff
@@ -200,6 +200,11 @@ def fields(include_deprecated: false)
       include_deprecated ? @fields : @fields.reject(&:deprecated?)
     end
 
+    def field_from_name(field_name, include_deprecated: false)
+      @fields_by_name ||= fields(include_deprecated: include_deprecated).map{ |field| [field.name, field]}.to_h
```

Comment:
This will memoize the hash based on the `include_deprecated` value of the first call and ignore it on subsequent calls.